### PR TITLE
Tmz 1.6.0 esoteric cow

### PR DESCRIPTION
**Pull request by t-m-z**
add interpreter/generator for esoteric programming language cow

### CODE REVIEW

**Comment by t-m-z** [comment]
merge conflicts solved

File: assets/i18n/de.json [hunk 1]
```diff
@@ -1204,6 +1204,21 @@
   "coords_averaging_stop": "Beende Messung",
   "coords_averaging_averagedcoordinate": "Gemittelte Koordinate\n(%s1 Messungen)",
   "coords_averaging_calculatedaccuracy": "Berechnete Genauigkeit",
+  "cow_title" : "Cow",
+  "cow_description" : "Esoterische Programmiersprache ähnlich Brainf**k, jedoch mit 12 Befehlen",
+  "cow_example" : "MoO moO MoO mOo MOO OOM MMM",
+  "cow_online_help_url" : "https://blog.gcwizard.net/manual/de/cow/02-wie-interpretiert-der-gc-wizard-ein-cow-programm/",
```
Comment (S-Man42) [inline assets/i18n/de.json:1210]: Not necessary anymore

File: lib/logic/tools/crypto_and_encodings/esoteric_programming_languages/cow.dart [hunk 2]
```diff
@@ -0,0 +1,324 @@
+//---------------------------------------------------------
+// COW Programming language
+// by BigZaphod sean@fifthace.com
+// https://github.com/BigZaphod/COW
+//
+// Licencse: Public Domain
+//
+// Interpreter Marco F. atomK
+// https://github.com/Atomk/C-COW-Interpreter/blob/master/cow-interpreter.c
```
Comment (S-Man42) [inline lib/logic/tools/crypto_and_encodings/esoteric_programming_languages/cow.dart:9]: Please add the licenses here as well.

File: lib/widgets/searchstrings/searchstrings_de.dart [hunk 3]
```diff
@@ -135,6 +135,7 @@ const SEARCHSTRING_DE_ESOTERICPROGRAMMINGLANGUAGE = 'esoterischeprogrammiersprac
 const SEARCHSTRING_DE_ESOTERIC_BEATNIK = '';
 const SEARCHSTRING_DE_ESOTERIC_BRAINFK = 'gehirn';
 const SEARCHSTRING_DE_ESOTERIC_CHEF = 'chef chefkoch kochrezepte rezepte kochen zutaten';
+final SEARCHSTRING_DE_ESOTERIC_COW = 'kuh muh';
```
Comment (S-Man42) [inline lib/widgets/searchstrings/searchstrings_de.dart:138]: kuehe rinder

File: lib/widgets/searchstrings/searchstrings_en.dart [hunk 4]
```diff
@@ -130,6 +130,7 @@ const SEARCHSTRING_EN_ESOTERICPROGRAMMINGLANGUAGE = 'esotericprogramminglanguage
 const SEARCHSTRING_EN_ESOTERIC_BEATNIK = '';
 const SEARCHSTRING_EN_ESOTERIC_BRAINFK = 'brain';
 const SEARCHSTRING_EN_ESOTERIC_CHEF = 'ingredients cooking cook recipes';
+final SEARCHSTRING_EN_ESOTERIC_COW = 'cow moo';
```
Comment (S-Man42) [inline lib/widgets/searchstrings/searchstrings_en.dart:133]: cow -> cows

File: lib/widgets/searchstrings/searchstrings_fr.dart [hunk 5]
```diff
@@ -124,6 +124,7 @@ const SEARCHSTRING_FR_ESOTERICPROGRAMMINGLANGUAGE = 'langage programmation exoti
 const SEARCHSTRING_FR_ESOTERIC_BEATNIK = '';
 const SEARCHSTRING_FR_ESOTERIC_BRAINFK = 'cerveau niquer';
 const SEARCHSTRING_FR_ESOTERIC_CHEF = 'cuisinier recettes cuisine';
+final SEARCHSTRING_FR_ESOTERIC_COW = 'vache';
```
Comment (S-Man42) [inline lib/widgets/searchstrings/searchstrings_fr.dart:127]: vache -> vaches

**Comment by S-Man42** [comment]
Can you please add the names of the original logic codes to the licenses page? :) (Maybe we still need to do this for all esoteric codes, don't we?)

File: lib/widgets/searchstrings/searchstrings_common.dart [hunk 6]
```diff
@@ -127,6 +127,7 @@ const SEARCHSTRING_COMMON_ESOTERIC_BEATNIK = 'beatnik cliff biffle';
 const SEARCHSTRING_COMMON_ESOTERIC_BRAINFK =
     'brainf**k brainfk urban mueller revolution9 detailedf**k  detailedfk colonoscopy k-on f**bees fkbees p***sscript psscript reversef**k reversefk binaryf**k binaryfk ternary kkenny speak morsef**k morsefk blub ook nak pikalang pikachu pokemon roadrunner zzz screamcode flufflepuff triplet uwu btjzxgquartfrqifjlv omam';
 const SEARCHSTRING_COMMON_ESOTERIC_CHEF = '';
+final SEARCHSTRING_COMMON_ESOTERIC_COW = 'sean heber';
```
Comment (S-Man42) [inline lib/widgets/searchstrings/searchstrings_common.dart:130]: moo oom mm

**Comment by S-Man42** [comment]
Found no big issues here! Great! Very well done!

File: assets/i18n/de.json (hunk 1 above)
Comment (t-m-z) [reply assets/i18n/de.json:1210]: removed

File: lib/logic/tools/crypto_and_encodings/esoteric_programming_languages/cow.dart (hunk 2 above)
Comment (t-m-z) [reply lib/logic/tools/crypto_and_encodings/esoteric_programming_languages/cow.dart:9]: done

File: lib/widgets/searchstrings/searchstrings_de.dart (hunk 3 above)
Comment (t-m-z) [reply lib/widgets/searchstrings/searchstrings_de.dart:138]: done

File: lib/widgets/searchstrings/searchstrings_en.dart (hunk 4 above)
Comment (t-m-z) [reply lib/widgets/searchstrings/searchstrings_en.dart:133]: done

File: lib/widgets/searchstrings/searchstrings_fr.dart (hunk 5 above)
Comment (t-m-z) [reply lib/widgets/searchstrings/searchstrings_fr.dart:127]: done

File: lib/widgets/searchstrings/searchstrings_common.dart (hunk 6 above)
Comment (t-m-z) [reply lib/widgets/searchstrings/searchstrings_common.dart:130]: done

**Comment by t-m-z** [comment]
> Can you please add the names of the original logic codes to the licenses page? :) (Maybe we still need to do this for all esoteric codes, don't we?)

Chef is already done
Malbolge is already done
Cow is added
Whitespace is added
Beatnik is added

**Comment by t-m-z** [comment]
> Found no big issues here! Great! Very well done!

Thanks a lot

File: lib/logic/tools/crypto_and_encodings/esoteric_programming_languages/cow.dart [hunk 7]
```diff
@@ -7,6 +7,28 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
 //
 // Generator   Frank Buss
 // https://frank-buss.de/cow.html
```
Comment (S-Man42) [inline lib/logic/tools/crypto_and_encodings/esoteric_programming_languages/cow.dart:34]: What about this guy? Did he agree that we take the code?

**Comment by t-m-z** [comment]
I did not found any information but will check again

Sent from MailDroid

-----Original Message-----

**Comment by S-Man42** [comment]
Yes, maybe you could ask him or something :) If there is no explicite license or permission, it's denied.

**Comment by t-m-z** [comment]
I got his permission - have a look at the GCWizard Mail Account

Sent from MailDroid

-----Original Message-----

**Pull request merged**